lib/AST/NameLookup.cpp

```diff
@@ -1169,6 +1191,11 @@ UnqualifiedLookup::UnqualifiedLookup(DeclName Name, DeclContext *DC,
   lookupInModule(&M, {}, Name, CurModuleResults, NLKind::UnqualifiedLookup,
                  resolutionKind, TypeResolver, DC, extraImports);
 
+  // Always perform name shadowing for type lookup.
+  if (options.contains(Flags::TypeLookup)) {
```

Comment:
This is iffy—it means that expression context lookup will behave differently than type context lookup. Or does that happen elsewhere?

Reply:
It does happen elsewhere. Most unqualified lookup goes through `TypeChecker::lookupUnqualified`, which (separately) triggers shadowing, so this is bringing things closer together in a duct-tape-and-paper-clip sort of way. There's more refactoring to do, but I'm holding out hope for this to be safe enough for Swift 5.0, hence the narrow applicability (to type lookup, to the Swift module, to non-member types);.